### PR TITLE
[omicron-package] add a way to show what Cargo commands will be run

### PR DESCRIPTION
**Pull request by sunshowers**
Add a way to see what commands will be run, without actually running them.

As part of this, refactor `CargoPlan` and change its meaning slightly -- it now
indicates the set of debug and release packages that would be built.

In the future, we'll be able to use these structures to add lints around which
features get built.

### CODE REVIEW

File: package/src/bin/omicron-package.rs [hunk 1]
```diff
@@ -105,128 +106,59 @@ struct Args {
...
+    let cargo_plan =
+        build_cargo_plan(&metadata, config.packages_to_build(), &features)?;
 
-impl<'a> CargoPlan<'a> {
-    async fn run(&self, log: &Logger) -> Result<()> {
-        if self.bins.is_empty() {
-            return Ok(());
-        }
+    let release_command = cargo_plan.release.build_command("build", true);
+    let debug_command = cargo_plan.debug.build_command("build", true);
```
Comment (smklein) [inline package/src/bin/omicron-package.rs:116]: The second argument to `build_command` is a boolean called `release` - should it be false?

File: package/src/lib.rs [hunk 2]
```diff
@@ -110,6 +111,7 @@ pub enum BuildCommand {
         #[clap(long)]
         intermediate: bool,
     },
+    /// List the build commands that will be run
```
Comment (smklein) [inline package/src/lib.rs:114]: Is this comment in the right spot? I don't think the `Package` command has changed in this PR

File: package/src/bin/omicron-package.rs (hunk 1 above)
Comment (sunshowers) [reply package/src/bin/omicron-package.rs:116]: Oh ugh... yeah. I'm going to make this an enum.

File: package/src/lib.rs (hunk 2 above)
Comment (sunshowers) [reply package/src/lib.rs:114]: No, whoops, not sure how it ended up there.

**Review by smklein — APPROVED** [review]

**Pull request merged**